data_store/messaging/tf_messaging.py

```diff
@@ -381,13 +409,17 @@ def _non_nullable_constraint_failure_message(self, validation_failure: NonNullab
             if column == "Financial Year 2022/21 - Financial Year 2025/26":
                 section = "Outcome Indicators (excluding footfall) / Footfall Indicator"
                 message = self.msgs.BLANK_ZERO
-                cell_index = self._get_cell_indexes_for_outcomes(validation_failure.failed_row)
+                cell_index = (
+                    (self._get_cell_indexes_for_outcomes(validation_failure.failed_row),)
+                    if validation_failure.failed_row is not None
+                    else ("",)
```

Comment:
In practice I think the `else` branch will never happen. All tests had `failed_row`.

If it would end up reaching it.. the code will throw an exception a bit later. Added a test for that case.

I will say that I don't know how `validation_failures` are built and which types has which values.

Is it weird? If there is no `failed_row` you get no `cell_index`. But Message class only likes `tuple[str, ...]`.